src/transformState.js

```diff
@@ -4,8 +4,30 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete state[keyToRemove];
+        }
+        break;
+
+      case 'clear':
+        for (const clearing in state) {
+          delete state[clearing];
+        }
+        break;
+
+      default:
+        continue;
```

Comment:
The `continue` statement in the `default` case of the switch is not necessary as it does nothing here. It could be removed for cleaner code.